nav2_collision_monitor/test/collision_monitor_node_test.cpp

```diff
@@ -850,6 +882,116 @@ TEST_F(Tester, testCeasePublishZeroVel)
+  parameter_msg->value.type = rcl_interfaces::msg::ParameterType::PARAMETER_BOOL;
+  parameter_msg->value.bool_value = false;
+  set_parameters_msg->parameters.push_back(*parameter_msg);
+  auto result_future = parameters_client_->async_send_request(set_parameters_msg).future.share();
+  ASSERT_TRUE(waitFuture(result_future, 2s));
+
+  // Check that robot does not stop when polygon is disabled
+  curr_time = cm_->now();
+  sendTransforms(curr_time);
+  publishScan(0.5, curr_time);
+  ASSERT_TRUE(waitData(0.5, 500ms, curr_time));
+  publishCmdVel(0.5, 0.2, 0.1);
+  ASSERT_TRUE(waitCmdVel(500ms));
+  ASSERT_NEAR(cmd_vel_out_->linear.x, 0.5, EPSILON);
+  ASSERT_NEAR(cmd_vel_out_->linear.y, 0.2, EPSILON);
+  ASSERT_NEAR(cmd_vel_out_->angular.z, 0.1, EPSILON);
+  // ASSERT_TRUE(waitActionState(500ms));
+  // ASSERT_EQ(action_state_->action_type, DO_NOTHING);
+  // ASSERT_EQ(action_state_->polygon_name, "");
+
+  // Stop Collision Monitor node
+  cm_->stop();
+}
+
+TEST_F(Tester, testSourceNotEnabled)
+{
+  // Set Collision Monitor parameters.
+  // Create a STOP polygon
+  setCommonParameters();
+  addPolygon("Stop", POLYGON, 1.0, "stop");
+  // Create a Scan source
+  addSource(SCAN_NAME, SCAN);
+  setVectors({"Stop"}, {SCAN_NAME});
+
+  // Start Collision Monitor node
+  cm_->start();
+
+  // Check that robot stops when source is enabled
+  rclcpp::Time curr_time = cm_->now();
+  sendTransforms(curr_time);
+  publishScan(0.5, curr_time);
+  ASSERT_TRUE(waitData(0.5, 500ms, curr_time));
+  publishCmdVel(0.5, 0.2, 0.1);
+  ASSERT_TRUE(waitCmdVel(500ms));
+  ASSERT_NEAR(cmd_vel_out_->linear.x, 0.0, EPSILON);
+  ASSERT_NEAR(cmd_vel_out_->linear.y, 0.0, EPSILON);
+  ASSERT_NEAR(cmd_vel_out_->angular.z, 0.0, EPSILON);
+  // ASSERT_TRUE(waitActionState(500ms));
+  // ASSERT_EQ(action_state_->action_type, STOP);
+  // ASSERT_EQ(action_state_->polygon_name, "Stop");
+
+  // Disable source by calling service
+  auto set_parameters_msg = std::make_shared<rcl_interfaces::srv::SetParameters::Request>();
+  auto parameter_msg = std::make_shared<rcl_interfaces::msg::Parameter>();
+  parameter_msg->name = std::string(SCAN_NAME) + ".enabled";
+  parameter_msg->value.type = rcl_interfaces::msg::ParameterType::PARAMETER_BOOL;
+  parameter_msg->value.bool_value = false;
+  set_parameters_msg->parameters.push_back(*parameter_msg);
+  auto result_future = parameters_client_->async_send_request(set_parameters_msg).future.share();
+  ASSERT_TRUE(waitFuture(result_future, 2s));
+
+  // Check that robot does not stop when source is disabled
+  curr_time = cm_->now();
+  sendTransforms(curr_time);
+  publishScan(0.5, curr_time);
+  ASSERT_TRUE(waitData(0.5, 500ms, curr_time));
+  publishCmdVel(0.5, 0.2, 0.1);
+  ASSERT_TRUE(waitCmdVel(500ms));
+  ASSERT_NEAR(cmd_vel_out_->linear.x, 0.5, EPSILON);
+  ASSERT_NEAR(cmd_vel_out_->linear.y, 0.2, EPSILON);
+  ASSERT_NEAR(cmd_vel_out_->angular.z, 0.1, EPSILON);
+  // ASSERT_TRUE(waitActionState(500ms));
```

Comment:
Why commented out?

Reply:
I have left this part as a comment so that you can easily identify it. As the final assertions of the test are no longer performed, I would prefer to have your instructions on whether I should keep the test incomplete or delete it. (And this part trigger an error as waitActionState is not defined)

Reply:
Why not add it?

```

bool Tester::waitActionState(const std::chrono::nanoseconds & timeout)
{
  rclcpp::Time start_time = cm_->now();
  while (rclcpp::ok() && cm_->now() - start_time <= rclcpp::Duration(timeout)) {
    if (action_state_) {
      return true;
    }
    rclcpp::spin_some(cm_->get_node_base_interface());
    std::this_thread::sleep_for(10ms);
  }
  return false;
}

```

Reply:
It was done outside of the commit [c2d84dfe671077bd76781ea356141b6d7ebb86f4](https://github.com/ros-navigation/navigation2/commit/c2d84dfe671077bd76781ea356141b6d7ebb86f4) so i didn't port it.
I will test it.

Reply:
Understood, thanks for the diligence!